src/cascadia/TerminalApp/Command.cpp

```diff
@@ -0,0 +1,191 @@
+    //   contain a "name" and "action", and optionally an "icon".
+    //   * "name": string|object - the name of the command to display in the
+    //     command palette. If this is an object, look for the "key" property,
+    //     and try to load the string from our resources instead.
+    //   * "action": string|object - A ShortcutAction, either as a name or as an
+    //     ActionAndArgs serialization. See ActionAndArgs::FromJson for details.
+    //     If this is null, we'll remove this command from the list of commands.
+    // Arguments:
+    // - json: the Json::Value to deserialize into a Command
+    // - warnings: If there were any warnings during parsing, they'll be
+    //   appended to this vector.
+    // Return Value:
+    // - the newly constructed Command object.
+    winrt::com_ptr<Command> Command::FromJson(const Json::Value& json,
+                                              std::vector<::TerminalApp::SettingsLoadWarnings>& warnings)
+    {
+        auto result = winrt::make_self<Command>();
+
+        // TODO GH#6644: iconPath not implemented quite yet. Can't seem to get
+        // the binding quite right. Additionally, do we want it to be an image,
+        // or a FontIcon? I've had difficulty binding either/or.
+
+        if (const auto actionJson{ json[JsonKey(ActionKey)] })
+        {
+            auto actionAndArgs = ActionAndArgs::FromJson(actionJson, warnings);
+
+            if (actionAndArgs)
+            {
+                result->_setAction(*actionAndArgs);
+            }
+            else
+            {
+                // Something like
+                //      { name: "foo", action: "unbound" }
+                // will _remove_ the "foo" command, by returning null here.
+                return nullptr;
+            }
+
+            result->_setName(_nameFromJsonOrAction(json, actionAndArgs));
+        }
+        else
+        {
+            // { name: "foo", action: null } will land in this case, which
+            // should also be used for unbinding.
+            return nullptr;
+        }
+
+        if (result->_Name.empty())
+        {
+            return nullptr;
+        }
+
+        return result;
+    }
+
+    // Function Description:
+    // - Attempt to parse all the json objects in `json` into new Command
+    //   objects, and add them to the map of commands.
+    // - If any parsed command has
+    //   the same Name as an existing command in commands, the new one will
+    //   layer on top of the existing one.
+    // Arguments:
+    // - commands: a map of Name->Command which new commands should be layered upon.
+    // - json: A Json::Value containing an array of serialized commands
+    // Return Value:
+    // - A vector containing any warnings detected while parsing
+    std::vector<::TerminalApp::SettingsLoadWarnings> Command::LayerJson(std::unordered_map<winrt::hstring, winrt::TerminalApp::Command>& commands,

```

Comment:
ParseAndLayerJsonCommands might be better
consider using a struct or pair that holds the 2 outputs instead of using the return + out param.